pkg/operator/ceph/disruption/nodedrain/reconcile.go

```diff
@@ -88,11 +88,26 @@ func (r *ReconcileNode) reconcile(request reconcile.Request) (reconcile.Result,
 	if err != nil {
 		return reconcile.Result{}, fmt.Errorf("could not list the osd pods: %+v", err)
 	}
+
+	// map with tolerations as keys and empty struct as values for uniqueness
+	uniqueTolerations := make(map[corev1.Toleration]struct{})
 	occupiedByOSD := false
 	for _, osdPod := range osdPodList.Items {
 		if osdPod.Spec.NodeName == request.Name {
 			occupiedByOSD = true
-			break
+			// get the osd tolerations
+			for _, osdToleration := range osdPod.Spec.Tolerations {
+				if osdToleration.Key == "node.kubernetes.io/unschedulable" {
+					labels := osdPod.GetLabels()
+					logger.Errorf(
```

Comment:
So we just log the error and that's it? Don't we want to return here?

Reply:
This just means that this particular OSD won't get drained at all as it tolerates the drain. We log error as this is bad practice (a drain might precede some other disruption), but there's not much else we can do. It doesn't effect the function of the drain canaries.

Reply:
Thanks!